bugbot/rules/perfalert_resolved_regression.py

```diff
@@ -0,0 +1,200 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from datetime import timedelta
+
+from libmozdata import utils as lmdutils
+from libmozdata.bugzilla import BugzillaUser
+
+from bugbot.bzcleaner import BzCleaner
+from bugbot.constants import BOT_MAIN_ACCOUNT
+
+DEFAULT_RESOLUTION_COMMENT = "No resolution comment provided."
+
+
+class PerfAlertResolvedRegression(BzCleaner):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.extra_email_info = {}
+        self.extra_ni = {}
+
+    def description(self):
+        return "PerfAlert regressions whose resolution has changed recently"
+
+    def columns(self):
+        return [
+            "id",
+            "summary",
+            "status",
+            "status_author",
+            "resolution",
+            "resolution_comment",
+            "resolution_previous",
+        ]
+
+    def get_extra_for_needinfo_template(self):
+        return self.extra_ni
+
+    def get_bz_params(self, date):
+        end_date = lmdutils.get_date_ymd("today")
+        start_date = end_date - timedelta(1)
+
+        fields = [
+            "id",
+            "history",
+            "comments.text",
+            "comments.creation_time",
+            "comments.author",
+        ]
+
+        # Find all bugs that have perf-alert, and regression in their keywords. Search
+        # for bugs that have been changed in the last day. Only look for bugs after
+        # October 1st, 2024 to prevent triggering comments on older performance regressions
+        params = {
+            "include_fields": fields,
+            "f3": "creation_ts",
+            "o3": "greaterthan",
+            "v3": "2024-10-01T00:00:00Z",
+            "f1": "regressed_by",
+            "o1": "isnotempty",
+            "f2": "keywords",
+            "o2": "allwords",
+            "v2": ["regression", "perf-alert"],
+            "f4": "resolution",
+            "o4": "changedafter",
+            "v4": start_date,
+            "f5": "resolution",
+            "o5": "changedbefore",
+            "v5": end_date,
+        }
+
+        return params
+
+    def should_needinfo(self, bug_comments, status_time):
+        # Check if the bugbot has already needinfo'ed on the bug since
+        # the last status change before making one
+        for comment in bug_comments[::-1]:
+            if lmdutils.get_date_ymd(comment["creation_time"]) <= status_time:
+                break
+
+            if comment["author"] == BOT_MAIN_ACCOUNT:
+                if (
+                    "comment containing a reason for why the performance regression"
+                    in comment["text"]
+                    or "could you provide a comment explaining the resolution?"
+                    in comment["text"]
```

Comment:
Why not use one phrase instead of two?

Reply:
I've changed this to one phrase now. I used two in case someone changes the needinfo comment without changing this code. It would lower the chances that it would stop working.